src/coreclr/jit/instr.cpp

```diff
@@ -838,6 +838,41 @@ CodeGen::OperandDesc CodeGen::genOperandDesc(GenTree* op)
                 assert(op->isContainedIntOrIImmed());
                 return OperandDesc(op->AsIntCon()->IconValue(), op->AsIntCon()->ImmedValNeedsReloc(compiler));
 
+            case GT_CNS_VEC:
+            {
+                switch (op->TypeGet())
+                {
+#if defined(FEATURE_SIMD)
+                    case TYP_LONG:
+                    case TYP_DOUBLE:
+                    case TYP_SIMD8:
+                    {
+                        // TYP_SIMD8 may get retyped to TYP_LONG or TYP_DOUBLE in lowering or morph
```

Comment:
`TODO-Cleanup: do not retype SIMD nodes, turn them into appropriate nodes of the other type instead.`